DEVELOP.md

```diff
@@ -0,0 +1,485 @@
+specification are rarely used.
+
+## Daffodil Development
+
+The Apache Software Foundation hosts the Apache Daffodil project on
+the following ASF infrastructure:
+
+- Daffodil's [issue
+  tracker](https://issues.apache.org/jira/projects/DAFFODIL/) is
+  hosted on JIRA
+- Daffodil's
+  [users](https://lists.apache.org/list.html?users@daffodil.apache.org),
+  [dev](https://lists.apache.org/list.html?dev@daffodil.apache.org),
+  and
+  [commits](https://lists.apache.org/list.html?commits@daffodil.apache.org)
+  mailing lists are hosted on Apache Pony Mail
+- Daffodil's [source code](https://github.com/apache/daffodil) is
+  hosted on GitHub
+- Daffodil's
+  [wiki](https://cwiki.apache.org/confluence/display/DAFFODIL/) is
+  hosted on Confluence
+- Daffodil's [website](https://daffodil.apache.org/) is hosted on
+  Apache with static content generated by Jekyll using jekyll-asciidoc
+  and asciidoctor-diagram plugins
+
+A good Daffodil developer has a GitHub account and two copies of
+Daffodil's source code repository, a reading copy directly cloned from
+the Apache repository and a working copy cloned from the developer's
+own fork of the Apache repository.  A good Daffodil developer also
+gets an account on the Apache Software Foundation's JIRA and
+Confluence servers and subscribes to all three Daffodil mailing lists.
+
+You can clone your reading copy directly from the Apache repository,
+but you should never make any changes in it.  The only commands you
+should ever run in your reading copy are `git pull`, `git log
+ORIG_HEAD..HEAD`, and `git diff ORIG_HEAD..HEAD` in order to see what
+changes have been made by other developers since your last pull from
+the Apache repository.  Your reading copy should remain an exact copy
+of the Apache source code repository at all times which you can use
+for reading or occasionally running `diff` commands between your
+reading copy and working copy.  To be safe, rename your reading copy
+from `daffodil` to `daffodil-asf` and never edit files or run `sbt` in
+it so there will be nothing to push to the Apache repository even if
+you accidentally run `git push` from the wrong source tree.
+
+You can set up your working copy by forking the Apache repository to
+your own GitHub account and then cloning your working copy from your
+fork.  Cloning your working copy from your fork instead of the Apache
+repository ensures that any changes you commit in your working copy
+will be pushed safely to only your own fork, not the Apache
+repository.
+
+### Code Style
+
+Daffodil mandates standard Scala formatting and its code generally
+keeps close to that formatting consistently.  No one has run an
+automatic Scala formatter on the codebase yet (there has not been much
+need since the code is already formatted pretty well) but
+[DAFFODIL-2133](https://issues.apache.org/jira/browse/DAFFODIL-2133)
+is waiting for someone willing to do that work.
+
+Daffodil mandates that at least 80% of new code and modified code
```

Comment:
We set the codecov target to `auto`, which I think just means a "patch coverage" must be the same or more than the base of the PR. Daffodil is just currently at about 80% coverage, so that's what PR's require. I"m not sure if that distinction is worth mentioning though. ~80% seems reasonable, especially since it will require a decent amount of work to make that number grow.